velox/functions/prestosql/window/tests/RankTest.cpp

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/exec/tests/utils/OperatorTestBase.h"
+#include "velox/exec/tests/utils/PlanBuilder.h"
+#include "velox/functions/prestosql/window/WindowFunctionsRegistration.h"
+#include "velox/vector/fuzzer/VectorFuzzer.h"
+
+using namespace facebook::velox::exec::test;
+
+namespace facebook::velox::window::test {
+
+namespace {
+
+class RankTest : public OperatorTestBase {
```

Comment:
Is there some copy paste between this test and RowNumberTest? Would it make sense to refactor to avoid code duplication? Maybe introduce WindowTestBase class?

Reply:
Yes, I will refactor to introduce a WindowTestBase class.

I'm doing another PR with more tests for both row_number and rank and will do the refactoring there. Just wanted to focus on only adding the new function in this PR.

Reply:
Would you refactor in this PR? It is a bit tedious to review copy-pasted code.